lib/ecto_ldap/adapter.ex

```diff
@@ -87,20 +80,25 @@ defmodule Ecto.Ldap.Adapter do
   end
   def construct_filter(_), do: nil
 
-  def extract_parameter(wheres) do
-    [{{_, _, [_ | parameter]}, _, _} | _] = extract_array(wheres)
-    case parameter do
-      {:&, [], [0]} -> []
-      _ -> to_string(hd(parameter))
-    end
+  defp base, do: Keyword.get(Application.get_env(:exldap, :settings), :base)
+
+  def construct_base(%{from: {from, _}}) do
+    {:base, "ou=" <> from <> "," <> base}
   end
+  def constuct_base(_), do: {:base, base}
 
-  def extract_criteria([]), do: []
-  def extract_criteria(wheres), do: hd(tl(extract_array(wheres)))
+  def construct_scope(_), do: {:scope, :eldap.wholeSubtree}
 
-  def extract_array([%Ecto.Query.QueryExpr{expr: {_, _, array}} | _tail]), do: array
+  def construct_attributes(_) do
+  end
 
-  def construct_scope, do: {:scope, :eldap.wholeSubtree}
+  #   def extract_parameter(wheres) do
+  #     [{{_, _, [_ | parameter]}, _, _} | _] = extract_array(wheres)
```

Comment:
I commented this code in anticipation of possibly re-using the gnarly pattern-matching referenced herein.  I'm hoping that given our earlier progress on the lispy-nature of Ecto, that this will be unnecessary.